src/home/user-record-util.c

```diff
@@ -1404,6 +1404,210 @@ int user_record_is_supported(UserRecord *hr, sd_bus_error *error) {
         return 0;
 }
 
+static int remove_safe_json_fields_regular(JsonVariant **v) {
```

Comment:
btw, i'd prefer if this function would only update *v when it succeeds. i.e. take a ref of *v into a local variable, mangle that. once everything is done unref *v, and overwrite it with the local version.

just as a matter of hygiene, as we usually say that functions should ideally either work or fail, but never fail and leave half-changed stuff around.

but doesn't matter much